Task/enable autoformatting

Graham42 opened the pull request:
Closes: https://github.com/englercj/tsd-jsdoc/issues/84

I've broken this down into separate commits so the configuration / code changes can be looked at separately from the formatting changes.

I tried to configure prettier to respect the existing code style. Let me know your thoughts or if there's any [other options](https://prettier.io/docs/en/options.html) you'd like me to set. If so I can rebase and re-do the formatting commit.

englercj commented on src/PropTree.ts:
```diff
@@ -1,28 +1,25 @@
 
-export interface IPropDesc
-{
+export interface IPropDesc {
```
I would like to keep the previous brace style, I don't like K&R style.

Graham42 replied:
This isn't an option with prettier.

englercj commented on src/PropTree.ts:
```diff
@@ -36,40 +33,35 @@ export class PropTree
 
-            if (!prop || !prop.name)
-                continue;
+            if (!prop || !prop.name) continue;
```
I'd rather keep these as separate lines.

Graham42 replied:
In order to keep these as separate lines with prettier would need to add in braces. 

Prettier [wont do that out of principle](https://prettier.io/docs/en/rationale.html#what-prettier-is-not-concerned-about) but there's an [eslint rule](https://eslint.org/docs/rules/curly) that can auto-add this.

englercj commented on src/PropTree.ts:
```diff
@@ -36,40 +33,35 @@ export class PropTree
+            if (!obj) {
+                warn(
+                    'Failed to find dot-notation property in map. This is likely a bug.',
+                );
```
This seems like a weird change.

Graham42 replied:
This is probably because of the line length that it gets wrapped.

englercj commented on src/create_helpers.ts:
```diff
@@ -1,67 +1,77 @@
+                warn(
+                    `Encountered child that is not a ${msg}, this is likely due to invalid JSDoc.`,
+                    child,
+                );
```
Maybe there is a max line length that is too short, or a minimum required params to make it do this? I don't mind splitting params to different lines, but this is happening too often.

Graham42 replied:
Depending the outcome of other comments, if this is the only this stopping us going forward, I can create a branch with a line length of 100 to compare.

englercj commented on src/create_helpers.ts:
```diff
@@ -1,67 +1,77 @@
+                !ts.isEnumDeclaration(child) &&
+                !ts.isModuleDeclaration(child) &&
+                !ts.isTypeAliasDeclaration(child) &&
+                !ts.isVariableStatement(child)
```
Definitely prefer my previous style of having the first item be on the same line as if, and have the operator precede the statement.

Graham42 replied:
This isn't an option with prettier

englercj commented on src/create_helpers.ts:
```diff
@@ -149,27 +169,30 @@ export function createFunction(doclet: IFunctionDoclet): ts.FunctionDeclaration
 
-    if (!doclet.memberof)
-        mods.push(declareModifier);
+    if (!doclet.memberof) mods.push(declareModifier);
```
control flow statements should always be followed by a newline.

Graham42 replied:
Same as above, would need to add in braces.

englercj commented on src/create_helpers.ts:
```diff
@@ -184,76 +207,79 @@ export function createClassMethod(doclet: IFunctionDoclet): ts.MethodDeclaration
-{
+export function createInterfaceMethod(
+    doclet: IFunctionDoclet,
+): ts.MethodSignature {
```
Yeah this is happening too much, the previous format was fine.

Graham42 replied:
Same as above, I can adjust the line length after resolving other comments.

englercj commented:
A lot of changed here, gave it a quick skim and commented on some of the major changes I saw. I think the goal of this PR would be to get the auto formatter to be making as few changes to the current code as possible, which means it is matching the current style as much as possible.

Graham42 commented:
I've updated the prettier config to 100 printWidth and added eslint with an autofix rule to add braces for if statements.

Overall, there a couple changes that we can't configure. If those are a deal breaker then kinda out of luck. My personal opinion is that the format matters less than the ability to autoformat and not have to think about it in the future.

englercj commented:
> Overall, there a couple changes that we can't configure. If those are a deal breaker then kinda out of luck. 

Yeah this is generally why I don't use auto-format utilities, they rarely can match the style I want.

> My personal opinion is that the format matters less than the ability to autoformat and not have to think about it in the future.

I can understand that point of view, but I don't share it.

Thanks for taking a stab at this, sorry it didn't work out. I'd prefer to keep the code as-is.
